Framework/API/src/ExperimentInfo.cpp

```diff
@@ -366,63 +483,47 @@ void ExperimentInfo::populateInstrumentParameters() {
-          double x = rtpValues.radius * std::sin(rtpValues.theta) *
-                     std::cos(rtpValues.phi);
-          paramMap.addPositionCoordinate(paramInfo->m_component, "x", x);
-          double y = rtpValues.radius * std::sin(rtpValues.theta) *
-                     std::sin(rtpValues.phi);
-          paramMap.addPositionCoordinate(paramInfo->m_component, "y", y);
-          double z = rtpValues.radius * std::cos(rtpValues.theta);
-          paramMap.addPositionCoordinate(paramInfo->m_component, "z", z);
+          rtpValues.phi = value;
+        if (rtpValues.haveRadius) {
+          V3D pos;
+          pos.spherical(rtpValues.radius, rtpValues.theta, rtpValues.phi);
+          paramMapForPosAndRot.addV3D(paramInfo->m_component,
+                                      ParameterMap::pos(), pos);
         }
       } else {
-        populateWithParameter(paramMap, paramN, *paramInfo, runData);
+        populateWithParameter(paramMap, paramMapForPosAndRot, paramN,
+                              *paramInfo, runData);
       }
     } catch (std::exception &exc) {
       g_log.information() << "Unable to add component parameter '"
                           << nameComp.first << "'. Error: " << exc.what();
       continue;
     }
   }
-}
-
-/**
- * Replaces current parameter map with a copy of the given map
- * Careful: Parameters that are stored in DetectorInfo are not automatically
- * handled.
- * @ pmap const reference to parameter map whose copy replaces the current
- * parameter map
- */
-void ExperimentInfo::replaceInstrumentParameters(
-    const Geometry::ParameterMap &pmap) {
-  populateIfNotLoaded();
-  m_spectrumInfoWrapper = nullptr;
-  m_detectorInfoWrapper = nullptr;
-  this->m_parmap.reset(new ParameterMap(pmap));
-  m_parmap->setDetectorInfo(m_detectorInfo);
-}
-
-/**
- * exchanges contents of current parameter map with contents of other map)
- * Careful: Parameters that are stored in DetectorInfo are not automatically
- * handled.
- * @ pmap reference to parameter map which would exchange its contents with
- * current map
- */
-void ExperimentInfo::swapInstrumentParameters(Geometry::ParameterMap &pmap) {
-  populateIfNotLoaded();
-  m_spectrumInfoWrapper = nullptr;
-  m_detectorInfoWrapper = nullptr;
-  this->m_parmap->swap(pmap);
-  m_parmap->setDetectorInfo(m_detectorInfo);
+  for (const auto &item : paramMapForPosAndRot) {
+    if (isPositionParameter(item.second->name())) {
+      const auto newRelPos = item.second->value<V3D>();
+      updatePosition(detectorInfo, *parInstrument, item.first, newRelPos);
+    } else if (isRotationParameter(item.second->name())) {
+      const auto newRelRot = item.second->value<Quat>();
+      updateRotation(detectorInfo, *parInstrument, item.first, newRelRot);
+    }
+    // Parameters for individual components (x,y,z) are ignored. ParameterMap
+    // did compute the correct compound positions and rotations internally.
```

Comment:
:+1: This seems solid, but is quite hard to follow. It took me a while to realise why it is important to add all this to the map that gets thrown away - could there be a comment added explaining this, perhaps above the `try` statement?